test/controllers/api/v2/users_controller_test.rb

```diff
@@ -0,0 +1,38 @@
+require 'test_helper'
+
+class Api::V2::UsersControllerTest < ActionController::TestCase
+
+  test "GET #me responds with 200" do
+    token = stub(:acceptable? => true)
+    user  = stub(:to_json => {})
+
+    @controller.stubs(:doorkeeper_token).returns(token)
+    @controller.stubs(:current_resource_owner).returns(user)
+
+    get :show, format: :json
+    assert_response 200
+  end
+
+  test "GET #me responds with 401 when unauthorized" do
+    # See: https://github.com/doorkeeper-gem/doorkeeper/blob/master/lib/doorkeeper/models/concerns/accessible.rb
+    # See: https://github.com/doorkeeper-gem/doorkeeper/blob/master/lib/doorkeeper/models/access_token_mixin.rb#L223
+    token = stub(:accessible? => false, :acceptable? => false)
+    user  = stub(:to_json => {})
+
+    @controller.stubs(:doorkeeper_token).returns(token)
+    @controller.stubs(:current_resource_owner).returns(user)
+
+    get :show, format: :json
+    assert_response 401
```

Comment:
Hei @zuzust! I finally had the opportunity to have a look at this. Great job.

Just expanding on what I had told you via telegram

> Lo del oauth el otro día le eché un vistazo rápido y me pareció molón. El único pero que le pondría es que los tests son muy mockeados y dependientes de la implementación. Pero vamos, que es una feature bien jodida de testar así que el mero hecho de que tenga tests ya mola. En resumen, que no me lo he podido mirar en detalle (descargar la rama y probar el código) pero en principio me mola.

As an example, I would just pass a real valid token here as a parameter for the action or in the request headers. That way the test is more resilient since it doesn't depend on implementation details, more realistic since it runs the real code and it serves better as documentation for the feature.

Just leaving it as a suggestion for the future in case the feature starts getting frequently used and evolves.